### PR TITLE
Add widget deletion and auto centering vertical

### PR DESCRIPTION
**Pull request by ppizarror**

### CODE REVIEW

File: docs/_source/advanced.rst [hunk 1]
```diff
@@ -106,17 +106,21 @@ added:
     class Menu(object):
         ...
 
-        def add_mymenu(self, params, **kwargs):
+        def add_mymenu(self, params, current=False, **kwargs):
```
Comment (anxuae) [inline docs/_source/advanced.rst:109]: add_mywidget

File: pygameMenu/menu.py [hunk 2]
```diff
@@ -96,6 +98,7 @@ def __init__(self,
                  height,
                  width,
                  title,
+                 auto_center_v=False,
```
Comment (anxuae) [inline pygameMenu/menu.py:101]: call it `center_content` to be consistent with associated method

File: docs/_source/advanced.rst (hunk 1 above)
Comment (ppizarror) [reply docs/_source/advanced.rst:109]: true!

File: pygameMenu/menu.py (hunk 2 above)
Comment (ppizarror) [reply pygameMenu/menu.py:101]: epic!

**Comment by ppizarror** [comment]
> I noticed that there is more methods with the `current` parameter, to avoid systematically add this parameter to public methods, it could be better to add a new method `get_current()`.
> 
> If the user want to work on the current menu, it call it first then work on the returned instance.
> The user code will be more comprehensive.

That could be done on another PR 🌞

**Comment by smilefab** [comment]
I found what I think is another bug. I tried to construct a minimal example:

    import pygame
    from pygame.locals import *
    import pygame_menu

    pygame.init()
    SCREEN = pygame.display.set_mode((600, 400))
    button_list = ['Option1', 'Option2']

    def placeholder():
        '''placeholder method'''

    def delete_widget():
        SUBMENU.remove_widget(SUBMENU.get_widget('Option2'))

    SUBMENU = pygame_menu.Menu(300, 400, 'Submenu',
                            theme=pygame_menu.themes.THEME_BLUE)
    for button in button_list:
        SUBMENU.add_button(button, placeholder, button_id=button)
    SUBMENU.add_button('Back', pygame_menu.events.RESET)


    MENU = pygame_menu.Menu(300, 400, 'Welcome',
                            theme=pygame_menu.themes.THEME_BLUE)
    MENU.add_button('Submenu', SUBMENU)
    MENU.add_button('Delete Widget', delete_widget)
    MENU.add_button('Quit', pygame_menu.events.EXIT)

    while True:
        MENU.mainloop(SCREEN)

I have two menus, a main menu `MENU` and a sub menu `SUBMENU`. In `MENU` I have a button to remove a button from `SUBMENU`.

When I open SUBMENU first and then delete the button, i.e.

test1

I get the following error:

test

When I first delete the button and then open `SUBMENU` no error occurs:

test2

I am pretty sure this has something to do with selecting a widget after removing the button. Also notice how in the recent example after opening `SUBMENU` no widget is selected.

If, instead of `SUBMENU.remove_widget(SUBMENU.get_widget('Option2'))`, I remove the first button with `SUBMENU.remove_widget(SUBMENU.get_widget('Option1'))`, i.e.

test3

the first widget IS selected.

Is this really a bug or am I just using the API wrong?

**Comment by ppizarror** [comment]
> I found what I think is another bug. I tried to construct a minimal example:
> 
> ```
> import pygame
> from pygame.locals import *
> import pygame_menu
> 
> pygame.init()
> SCREEN = pygame.display.set_mode((600, 400))
> button_list = ['Option1', 'Option2']
> 
> def placeholder():
>     '''placeholder method'''
> 
> def delete_widget():
>     SUBMENU.remove_widget(SUBMENU.get_widget('Option2'))
> 
> SUBMENU = pygame_menu.Menu(300, 400, 'Submenu',
>                         theme=pygame_menu.themes.THEME_BLUE)
> for button in button_list:
>     SUBMENU.add_button(button, placeholder, button_id=button)
> SUBMENU.add_button('Back', pygame_menu.events.RESET)
> 
> 
> MENU = pygame_menu.Menu(300, 400, 'Welcome',
>                         theme=pygame_menu.themes.THEME_BLUE)
> MENU.add_button('Submenu', SUBMENU)
> MENU.add_button('Delete Widget', delete_widget)
> MENU.add_button('Quit', pygame_menu.events.EXIT)
> 
> while True:
>     MENU.mainloop(SCREEN)
> ```
> 
> I have two menus, a main menu `MENU` and a sub menu `SUBMENU`. In `MENU` I have a button to remove a button from `SUBMENU`.
> 
> When I open SUBMENU first and then delete the button, i.e.
> 
> test1
> 
> I get the following error:
> 
> test
> 
> When I first delete the button and then open `SUBMENU` no error occurs:
> 
> test2
> 
> I am pretty sure this has something to do with selecting a widget after removing the button. Also notice how in the recent example after opening `SUBMENU` no widget is selected.
> 
> If, instead of `SUBMENU.remove_widget(SUBMENU.get_widget('Option2'))`, I remove the first button with `SUBMENU.remove_widget(SUBMENU.get_widget('Option1'))`, i.e.
> 
> test3
> 
> the first widget IS selected.
> 
> Is this really a bug or am I just using the API wrong?

can you post this issue as a new issue to track it? I'll take a look at it

**Pull request merged**